src/phoenix/db/bulk_inserter.py

```diff
@@ -86,28 +91,60 @@ def last_updated_at(self, project_rowid: Optional[ProjectRowId] = None) -> Optio
 
     async def __aenter__(
         self,
-    ) -> Tuple[Callable[[Span, str], Awaitable[None]], Callable[[pb.Evaluation], Awaitable[None]]]:
+    ) -> Tuple[
+        Callable[[Span, str], Awaitable[None]],
+        Callable[[pb.Evaluation], Awaitable[None]],
+        Callable[[DataModification], Awaitable[None]],
+    ]:
         self._running = True
+        self._jobs = Queue()
         self._task = asyncio.create_task(self._bulk_insert())
-        return self._queue_span, self._queue_evaluation
+        return (
+            self._queue_span,
+            self._queue_evaluation,
+            self._enqueue_for_transaction,
+        )
 
     async def __aexit__(self, *args: Any) -> None:
+        self._jobs = None
         self._running = False
 
+    async def _enqueue_for_transaction(self, job: DataModification) -> None:
+        await cast("Queue[DataModification]", self._jobs).put(job)
```

Comment:
```suggestion
        await cast(Queue[DataModification], self._jobs).put(job)
```

I think the quotes are unnecessary here.

Reply:
Can we get away with a non-optional type for `self._jobs` to avoid needed to recast the type in this way?

Reply:
We can't, because it has to bee initialized inside an event loop, but we need to have the `bulk_inserter` before the loop starts.